### PR TITLE
Unpin frontend_server_client and roll packages

### PR DESCRIPTION
**Pull request by derekxu16**

### CODE REVIEW

**Comment by derekxu16** [comment]
https://github.com/dart-lang/sdk/commit/69b801f73f7b26adb661257ecab862eca6dcf7dd has made it so that AOT snapshots now get closed as soon as they're done being read. This means that bumping `package:frontend_server_client` no longer breaks `batch_entrypoint_test`. Is it ok to unpin `package:frontend_server_client` now, @christopherfujino?

**Comment by christopherfujino** [comment]
> [dart-lang/sdk@69b801f](https://github.com/dart-lang/sdk/commit/69b801f73f7b26adb661257ecab862eca6dcf7dd) has made it so that AOT snapshots now get closed as soon as they're done being read. This means that bumping `package:frontend_server_client` no longer breaks `batch_entrypoint_test`. Is it ok to unpin `package:frontend_server_client` now, @christopherfujino?

Yes, I was just waiting for that commit to roll, thanks!

File: packages/flutter_tools/pubspec.yaml [hunk 1]
```diff
@@ -68,7 +68,7 @@ dependencies:
   vm_service: 14.2.0
 
   standard_message_codec: 0.0.1+4
-  frontend_server_client: 3.2.0 # exact pin because of https://github.com/flutter/flutter/issues/146164
+  frontend_server_client: 4.0.0
```
Comment (christopherfujino) [inline packages/flutter_tools/pubspec.yaml:71]: This no longer needs to be exactly pinned, can you actually just delete this line and then re-run `flutter update-packages --force-upgrade`? It should add this same version but with the `# THIS LINE IS AUTOGENERATED...` comment

**Review by christopherfujino — APPROVED** [review]
LGTM, but please remove the exact pin of frontend_server_client before merging

File: packages/flutter_tools/pubspec.yaml (hunk 1 above)
Comment (derekxu16) [reply packages/flutter_tools/pubspec.yaml]: Ah, right. Done.

**Pull request merged**